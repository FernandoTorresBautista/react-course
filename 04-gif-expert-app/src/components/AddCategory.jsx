import { useState } from "react";
import { PropTypes } from "prop-types";

export const AddCategory = ( {onNewCategorie} ) => {

    const [inputValue, setInputValue] = useState('');

    const onInputChange = (event) => {
        setInputValue(event.target.value);
    }

    const onSubmit = (event) => {
        event.preventDefault();
        if (inputValue.trim().length <= 1 ) return;
        onNewCategorie(inputValue.trim());
        setInputValue('');
    }

    return (
        <form onSubmit={(event) => onSubmit(event)} aria-label="form">
            <input
                type="text"
                placeholder="search gifs"
                value={inputValue}
                onChange={ (event) => onInputChange(event) }
            />
        </form>
    )
}

AddCategory.propTypes = {
    onNewCategorie: PropTypes.func.isRequired,
}