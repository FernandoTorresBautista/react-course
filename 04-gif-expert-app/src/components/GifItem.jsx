import PropTypes from 'prop-types';

export const GifItem = ({ id, title, url }) => {

    return (
        <div className="card">
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    )
}

GifItem.propTypes = {
    // id: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}

// GifItem.defaultProps = {
//     id: 0,
//     title: 'No hay t√≠tulo',
//     url: "No existe url",
// }
