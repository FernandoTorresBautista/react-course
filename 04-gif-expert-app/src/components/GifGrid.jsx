import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifItem } from "./GifItem";
import { PropTypes } from "prop-types";

export const GifGrid = ({ category }) => {

    const {images, isLoading} = useFetchGifs( category );

    return (
        <>
            <h3>{category}</h3>
            {
                // logic and
                isLoading && (<h2>Cargando...</h2>)
            }

            <div className="card-grid">
                {
                    images.map( (img) => (
                            <GifItem 
                                key={img.id}
                                // with {...img} spread img properties
                                {...img}
                            />
                        )) 
                }
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
}
