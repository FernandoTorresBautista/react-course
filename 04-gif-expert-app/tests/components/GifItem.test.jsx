import { render, screen, fireEvent } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('testing on <GifItem />', () => {
    
    const id = 100;
    const title = "My title";
    const url = "http://localhost/not-exit";


    test('should match with the snapshot', () => {
        
        const {container} = render( <GifItem title={ title } url={ url } /> );
        expect( container ).toMatchSnapshot();

    });

    test('should show the image with the URL and ALT', () => {
        
        render( <GifItem title={ title } url={ url } /> );
        const {src, alt} = screen.getByRole("img");
        expect( src ).toBe( url );
        expect( alt ).toBe( title );

    });

});
