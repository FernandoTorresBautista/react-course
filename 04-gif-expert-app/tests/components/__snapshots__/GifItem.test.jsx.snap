// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing on <GifItem /> should match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="My title"
      src="http://localhost/not-exit"
    />
    <p>
      My title
    </p>
  </div>
</div>
`;
